pkg/client/interfaces.go

```diff
@@ -91,21 +92,9 @@ type GenConfig struct {
 
 // Validate checks the config to determine if it is valid.
 func (gc *GenConfig) Validate() error {
-	if gc.E2EConfig == nil {
-		return errors.New("nil E2EConfig provided")
-	}
-
 	return nil
 }
 
-// E2EConfig is the configuration of the E2E tests.
-type E2EConfig struct {
```

Comment:
No more e2econfig required at all. Everything is just transforms on the plugin; changing image, changing env.